addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/http/IhcConnectionPool.java

```diff
@@ -0,0 +1,146 @@
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Custom HTTP connection pool, which install all-trusting trust manager.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class IhcConnectionPool {
+
+    private final Logger logger = LoggerFactory.getLogger(IhcConnectionPool.class);
+
+    /**
+     * Controller TLS certificate is self signed, which means that certificate
+     * need to be manually added to java key store as a trusted certificate.
+     * This is special SSL context which will be configured to trust all
+     * certificates and manual work is not required.
+     */
+    private SSLContext sslContext;
+
+    /** Holds and share cookie information (session id) from authentication procedure */
+    private CookieStore cookieStore;
+
+    private HttpClientBuilder httpClientBuilder;
+    private HttpClientContext localContext;
+
+    public IhcConnectionPool() {
+        init();
+    }
+
+    private void init() {
+        // Create a local instance of cookie store
+        cookieStore = new BasicCookieStore();
+
+        // Create local HTTP context
+        localContext = HttpClientContext.create();
+
+        // Bind custom cookie store to the local context
+        localContext.setCookieStore(cookieStore);
+
+        httpClientBuilder = HttpClientBuilder.create();
+
+        // Setup a Trust Strategy that allows all certificates.
+
+        logger.debug("Initialize SSL context");
+
+        // Create a trust manager that does not validate certificate chains, but accept all.
+        TrustManager[] trustAllCerts = new TrustManager[] { new X509TrustManager() {
+
+            @Override
+            public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                return null;
+            }
+
+            @Override
+            public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+            }
+
+            @Override
+            public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType) {
+                logger.trace("Trusting server cert: " + certs[0].getIssuerDN());
+            }
+        } };
+
+        // Install the all-trusting trust manager
+
+        try {
+            // Controller supports only SSLv3 and TLSv1
+            sslContext = SSLContext.getInstance("TLSv1");
```

Comment:
Does this leak out, will it influence other connections made by other bindings in the same JVM. @openhab/2-x-add-ons-maintainers does one of you know that?

Reply:
No,it should not, it's affect only to local connection factory. I have tested this in OH1 binding many years ago.

This would be disaster: SSLContext.setDefault(sslContext);

Reply:
Great :-)